### DISCUSSION TITLE
How can I have a variant that accepts any string (w/ TypeScript)?

### DISCUSSION THREAD

**Discussion by MarkLyck** [joe-bell/cva#124] [Q&A]
I have a simple variant like this:

```ts
const example = cva([], {
  variants: {
      type: {
        a: 'bg-blue-500 text-blue-600',
        b: 'bg-yellow-500 text-yellow-600',
        c: 'bg-red-500 text-red-600',
      }
  }
})
```

This creates a type that says variants of `type` MUST be `a | b | c | null | undefined`

but I want to be able to pass an arbitrary string, if it doesn't match any existing variant, it shouldn't do anything.

But everything I've tried I'm getting an error: `Type 'string' is not assignable to type '"a" | "b" | "c" | null | undefined'.`

unless I override it with `as any` but I'd rather not add `any`'s to my codebase.

Is there any way to make the type for this particular variant less strong? So it can accept an arbitrary string?

**Comment by joe-bell** [comment]
Could you use `as`?

`(arbitraryValue) => example({ type: arbitraryValue as keyof VariantProps<typeof example>['type'] })`
